### DISCUSSION TITLE
Self-holding and open-sourced, which comes first?

### DISCUSSION THREAD

**Discussion by forfudan** [modular/modular#3049] [Q&A] [answered]
As far as I know, the Mojo compiler is written in C++. I am wondering whether Mojo will first become self-holding (compiler written in Mojo), or the compiler will first be open-sourced.

May I ask about the planning on this?

**Comment by jackos** [comment, marked as answer]
Hi @forFudan yes the team would like to make Mojo self hosted in the long term, which will also be a good opportunity to open source more. Note that there aren't any solid plans yet though, this is a long-term goal.

**Reply by forfudan** [reply]
Thank you very much for your answer!
